scripts/product_generators.rb

```diff
@@ -126,8 +126,26 @@ def get_operation(n, algebraic_structure, operation)
   "override def #{operation}(#{method_params}) = { val lTuple = unapply(l).get; val rTuple = unapply(r).get; #{values_commaed} }"
 end
 
+def get_sumoption(n, bufferSize)
+  # Example: "tsemigroup.sumOption(items.iterator.map(_._1), "
+  values_commaed = TYPE_SYMBOLS.first(n).each_with_index.map do |t, i|
+    "#{t.downcase}semigroup.sumOption(iter.map(_._#{i+1})).get"
```

Comment:
I think we do want the outer copy of the iterable, but not the second one that gets made to do this map. For this one I do think we want to pass an iterator to sumOption.